include/swift/AST/DiagnosticsParse.def

```diff
@@ -131,17 +131,17 @@ ERROR(lex_unicode_escape_braces,none,
 ERROR(lex_illegal_multiline_string_start,none,
       "multi-line string literal content must begin on a new line", ())
 ERROR(lex_illegal_multiline_string_end,none,
-      "multi-line string literal content must not be on the delimiter's line", ())
-ERROR(lex_insufficient_string_indent,none,
-      "string literal content is insufficiently indented", ())
-ERROR(lex_unexpected_tab_in_string_indent,none,
-      "string literal content is indented with a tab where a space is expected", ())
-ERROR(lex_unexpected_space_in_string_indent,none,
-      "string literal content is indented with a space where a tab is expected", ())
+      "multi-line string literal closing delimiter must begin on a new line", ())
+ERROR(lex_inconsistent_string_indent,none,
+      "line does not start with same indentation as closing delimiter of multi-line string literal", ())
+NOTE(note_insufficient_string_indent,none,
+      "unexpectedly found non-indentation character here", ())
+NOTE(note_unexpected_char_in_string_indent,none,
+      "unexpected %select{space|tab}0 in indentation here", (bool))
+NOTE(note_matching_indentation_char_here,none,
+      "should match %select{space|tab}0 in same position before closing delimiter", (bool))
```

Comment:
That's a fair point about flagging both locations.

My thinking here is that, for better or worse, Swift error messages tend to be pretty terse. On the one hand, it's nice that you're making this message more helpful. On the other, I worry that the first reaction when a user triggers this error is, "Gah, what have I done? All these messages!"

Keep in mind that a user could have pasted 100 lines and indented only the closing delimiter inconsistently. Unless I'm mistaken, in that scenario, your design could generate 100 errors and 300 notes for a single misplaced tab! My instinct is that this will actually give people a distinctly negative impression of multiline literals, even though you're only trying to make them easier to use.

If we must have _three_ separate messages for each poorly indented line, then at least we can make each message more bite-sized. How do you feel about something like this:

* "error: unexpected indentation in multi-line string literal"
* "note: change indentation of this line to match closing delimiter"
* "note: indentation of preceding lines must start with these characters"--_emit this only once per literal_